src/database/cache/mod.rs

```diff
@@ -0,0 +1,50 @@
+//pub mod project_cache;
+//pub mod project_query_cache;
+#[macro_export]
+macro_rules! generate_cache {
+    ($name:ident,$id:ty, $val:ty, $cache_name:ident, $mod_name:ident, $getter_name:ident, $setter_name:ident) => {
+        pub mod $mod_name {
+            use cached::async_mutex::Mutex;
+            use cached::{Cached, SizedCache};
+            use lazy_static::lazy_static;
+            use log::info;
+            lazy_static! {
+                static ref $cache_name: Mutex<SizedCache<$id, $val>> =
+                    Mutex::new(SizedCache::with_size(400));
+            }
+
+            pub async fn $getter_name<'a>(id: $id) -> Option<$val> {
+                let mut cache = $cache_name.lock().await;
+                info!("Querying cache {} for {}", stringify!($name), id);
```

Comment:
Info calls are enabled even with debig false, right? Is there a log debug call?

Reply:
Currently, labrinth lacks a way to control the log levels in the code, as displayed in the line present in the main.rs file:
```rust
env_logger::Builder::from_env(Env::default().default_filter_or("info")).init();
```
So there is a debug macro in the log crate, but we won't be able to get them to display.